src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { of } from 'rxjs';
+
+import { ArtemisTestModule } from '../../../test.module';
+import { MetricsComponent } from 'app/admin/metrics/metrics.component';
+import { MetricsService } from 'app/admin/metrics/metrics.service';
+import { Metrics, ThreadDump } from 'app/admin/metrics/metrics.model';
+
+describe('MetricsComponent', () => {
+    let comp: MetricsComponent;
+    let fixture: ComponentFixture<MetricsComponent>;
+    let service: MetricsService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [MetricsComponent],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(MetricsComponent);
+                comp = fixture.componentInstance;
+                service = fixture.debugElement.injector.get(MetricsService);
+            });
+    });
+
+    it('should call refresh on init', () => {
+        const mockMetrics = {};
+        const mockThreadDump = { threads: [] };
+        jest.spyOn(service, 'getMetrics').mockReturnValue(of(mockMetrics as Metrics));
+        jest.spyOn(service, 'threadDump').mockReturnValue(of(mockThreadDump as ThreadDump));
+        expect(comp.updatingMetrics).toBe(true);
+        comp.ngOnInit();
+        expect(service.getMetrics).toHaveBeenCalled();
+        expect(service.threadDump).toHaveBeenCalled();
```

Comment:
Would it be possible to specify the number of calls?

Reply:
Added in b28d473